deps.macos-12.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+# This script is used to install dependencies on MacOS 12.0 or newer
+# botan v3
+brew install googletest spdlog c-ares openssl@3  botan
```

Comment:
DITTO (and other dep installing scripts), how are we going to keep these from getting out-of-date?